Can't start the app after generating a project with MSSQL : Cannot insert explicit value when IDENTITY_INSERT is set to OFF

##### **Overview of the issue**

After generating a project with MSSQL, and starting the app with IntelliJ, we get the following error : 

```
t.j.c.liquibase.AsyncSpringLiquibase     : Liquibase could not start correctly, your database is NOT ready: liquibase.exception.MigrationFailedException: Migration failed for change set config/liquibase/changelog/00000000000000_initial_schema.xml::00000000000001::jhipster:
     Reason: liquibase.exception.DatabaseException: java.sql.BatchUpdateException: Cannot insert explicit value for identity column in table 'jhi_user' when IDENTITY_INSERT is set to OFF.
```

Even though in the `00000000000000_initial_schema.xml` we have the following properties :

```xml
        <ext:loadData
                  file="config/liquibase/data/user.csv"
                  separator=";"
                  identityInsertEnabled="true"
                  tableName="jhi_user">
            <column name="activated" type="boolean"/>
            <column name="created_date" type="timestamp"/>
        </ext:loadData>
```

With the `identityInsertEnabled`, set to `true` (generated like this by default), we still can't start the app properly because liquibase can't manage to load the custom data in the CSV.

##### **Motivation for or Use Case**

Liquibase does not work properly right after generating the project, and we did not add any custom code.

##### **Reproduce the error**

Using Java 15, generating a project with MSSQL (with the JHipster configuration given below), and then running the project with IntelliJ.

##### **JHipster Version(s)**

JHipster 7.0.1

##### **JHipster configuration**

```json
{
  "generator-jhipster": {
    "entitySuffix": "Entity",
    "dtoSuffix": "JhiDto",
    "blueprints": [],
    "otherModules": [],
    "applicationType": "monolith",
    "baseName": "myApp",
    "jhipsterVersion": "7.0.1",
    "skipClient": false,
    "skipServer": false,
    "skipUserManagement": false,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "jhiPrefix": "jhi",
    "testFrameworks": ["cypress", "gatling", "cucumber"],
    "pages": [],
    "creationTimestamp": 1621254883770,
    "serviceDiscoveryType": false,
    "reactive": true,
    "authenticationType": "jwt",
    "packageName": "my.app",
    "serverPort": "8080",
    "cacheProvider": "no",
    "enableHibernateCache": false,
    "databaseType": "sql",
    "devDatabaseType": "mssql",
    "prodDatabaseType": "mssql",
    "buildTool": "maven",
    "serverSideOptions": ["enableSwaggerCodegen:true"],
    "websocket": false,
    "searchEngine": false,
    "messageBroker": false,
    "enableSwaggerCodegen": true,
    "clientFramework": "angularX",
    "withAdminUi": true,
    "clientTheme": "none",
    "enableTranslation": true,
    "nativeLanguage": "fr",
    "packageFolder": "my/app",
    "jwtSecretKey": "My JWT Secret Key",
    "clientPackageManager": "npm",
    "clientThemeVariant": "",
    "languages": ["fr", "en"]
  }
}
```

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

We do not have the generated `.jhipster` directory since it is still a blank project.

##### **Browsers and Operating System**

Ubuntu 20.04

- [ x ] Checking this box is mandatory (this is just to show you read everything)

> Is it even possible to use webflux with MSSQL ?

It looks like there is an R2DBC driver for MSSQL, so it _should_ be possible.

https://github.com/r2dbc/r2dbc-mssql

I tried generating an app with your `.yo-rc.json` and then searched its `pom.xml` for `r2dbc-mssql`. It does exist.

```xml
<dependency>
  <groupId>io.r2dbc</groupId>
  <artifactId>r2dbc-mssql</artifactId>
</dependency>
```

Next, I tried to start it up by starting MSSQL in Docker.

```
docker-compose -f src/main/docker/mssql.yml up
```

Then, I started the app with `./mvnw`. I got a number of compilation errors on startup, which is different from your error.

```
[INFO] Compiling 66 source files to /Users/mraible/mssql/target/classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /Users/mraible/mssql/src/main/java/my/app/repository/UserRepository.java:[34,57] cannot find symbol
  symbol: class User
[ERROR] /Users/mraible/mssql/src/main/java/my/app/repository/UserRepository.java:[65,57] cannot find symbol
  symbol: class User
[ERROR] /Users/mraible/mssql/src/main/java/my/app/service/UserService.java:[355,17] cannot find symbol
  symbol:   class User
  location: class my.app.service.UserService
```

Renaming `my.app.domain.UserEntity` to `my.app.domain.User` gets me a bit further.

```
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 
'tokenProvider' defined in file [/Users/mraible/mssql/target/classes/my/app/security/jwt/TokenProvider.class]: 
Bean instantiation via constructor failed; nested exception is org.springframework.beans.BeanInstantiationException: 
Failed to instantiate [my.app.security.jwt.TokenProvider]: Constructor threw exception; 
nested exception is io.jsonwebtoken.io.DecodingException: Illegal base64 character: ' '
```

I updated the `base64-secret` secret's value in `application-dev.yml` and tried again. This time, it started but I'm unable to log in.

```
org.springframework.transaction.CannotCreateTransactionException: Could not open R2DBC Connection for transaction;
 nested exception is java.net.UnknownHostException: failed to resolve 'localhost:1433;database=myApp' after 2 queries
	at org.springframework.r2dbc.connection.R2dbcTransactionManager.lambda$null$5(R2dbcTransactionManager.java:226)
	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException:
```

--- a/generators/generator-base-private.js
+++ b/generators/generator-base-private.js
@@ -1281,7 +1281,11 @@ module.exports = class JHipsterBasePrivateGenerator extends Generator {
     } else if (databaseType === ORACLE) {
       dbcUrl = `${protocol}:oracle:thin:@${options.hostname}:1521:${options.databaseName}`;
     } else if (databaseType === MSSQL) {
-      dbcUrl = `${protocol}:sqlserver://${options.hostname}:1433;database=${options.databaseName}`;
+      if (protocol === 'r2dbc') {
+        dbcUrl = `${protocol}:mssql://${options.hostname}:1433/${options.databaseName}`;
+      } else {
+        dbcUrl = `${protocol}:sqlserver://${options.hostname}:1433;database=${options.databaseName}`;
+      }
     } else if (databaseType === H2_DISK) {
       if (!options.localDirectory) {
         throw new Error(`'localDirectory' option should be provided for ${databaseType} databaseType`);

--- a/generators/server/templates/build.gradle.ejs
+++ b/generators/server/templates/build.gradle.ejs
@@ -535,13 +535,12 @@ dependencies {
     <%_ } _%>
     <%_ if (prodDatabaseType === 'mssql') { _%>
     implementation "com.microsoft.sqlserver:mssql-jdbc"
-    implementation "com.github.sabomichal:liquibase-mssql"
-        <%_ if (!reactive) { _%>
-    liquibaseRuntime  "com.microsoft.sqlserver:mssql-jdbc"
-    liquibaseRuntime "com.github.sabomichal:liquibase-mssql"
-        <%_ } else { _%>
+    implementation "org.liquibase.ext:liquibase-mssql:<%= LIQUIBASE_VERSION %>"
+        <%_ if (reactive) { _%>
     implementation "io.r2dbc:r2dbc-mssql"
         <%_ } _%>
+    liquibaseRuntime "com.microsoft.sqlserver:mssql-jdbc"
+    liquibaseRuntime "org.liquibase.ext:liquibase-mssql:<%= LIQUIBASE_VERSION %>"
     <%_ } _%>
     implementation "org.mapstruct:mapstruct:${mapstructVersion}"
     annotationProcessor "org.mapstruct:mapstruct-processor:${mapstructVersion}"

--- a/generators/server/templates/pom.xml.ejs
+++ b/generators/server/templates/pom.xml.ejs
@@ -476,8 +476,9 @@
             <artifactId>mssql-jdbc</artifactId>
         </dependency>
         <dependency>
-            <groupId>com.github.sabomichal</groupId>
+            <groupId>org.liquibase.ext</groupId>
             <artifactId>liquibase-mssql</artifactId>
+            <version>${liquibase.version}</version>
         </dependency>
     <%_ if (reactive) { _%>
         <dependency>

--- a/generators/server/templates/src/main/java/package/repository/UserRepository.java.ejs
+++ b/generators/server/templates/src/main/java/package/repository/UserRepository.java.ejs
@@ -244,23 +244,23 @@ public interface UserRepository extends <% if (databaseType === 'sql') { %>JpaRe
 }
 <%_ } else if (databaseType === 'sql' && reactive) { _%>
 @Repository
-public interface UserRepository extends R2dbcRepository<User, <% if (authenticationType === 'oauth2') { %>String<% } else { %>Long<% } %>>, UserRepositoryInternal {
+public interface UserRepository extends R2dbcRepository<<%= asEntity('User') %>, <% if (authenticationType === 'oauth2') { %>String<% } else { %>Long<% } %>>, UserRepositoryInternal {
 
     <%_ if (authenticationType !== 'oauth2') { _%>
-    Mono<User> findOneByActivationKey(String activationKey);
+    Mono<<%= asEntity('User') %>> findOneByActivationKey(String activationKey);
 
-    Flux<User> findAllByActivatedIsFalseAndActivationKeyIsNotNullAndCreatedDateBefore(LocalDateTime dateTime);
+    Flux<<%= asEntity('User') %>> findAllByActivatedIsFalseAndActivationKeyIsNotNullAndCreatedDateBefore(LocalDateTime dateTime);
 
-    Mono<User> findOneByResetKey(String resetKey);
+    Mono<<%= asEntity('User') %>> findOneByResetKey(String resetKey);
 
-    Mono<User> findOneByEmailIgnoreCase(String email);
+    Mono<<%= asEntity('User') %>> findOneByEmailIgnoreCase(String email);
 
     <%_ } _%>
-    Mono<User> findOneByLogin(String login);
+    Mono<<%= asEntity('User') %>> findOneByLogin(String login);
 
-    Flux<User> findAllByIdNotNull(Pageable pageable);
+    Flux<<%= asEntity('User') %>> findAllByIdNotNull(Pageable pageable);
 
-    Flux<User> findAllByIdNotNullAndActivatedIsTrue(Pageable pageable);
+    Flux<<%= asEntity('User') %>> findAllByIdNotNullAndActivatedIsTrue(Pageable pageable);
 
     Mono<Long> count();
 
@@ -281,20 +281,20 @@ interface DeleteExtended<T> {
 }
 
 <%_ } _%>
-interface UserRepositoryInternal<% if (authenticationType !== 'oauth2') { %> extends DeleteExtended<User><% } %> {
+interface UserRepositoryInternal<% if (authenticationType !== 'oauth2') { %> extends DeleteExtended<<%= asEntity('User') %>><% } %> {
 
-    Mono<User> findOneWithAuthoritiesByLogin(String login);
+    Mono<<%= asEntity('User') %>> findOneWithAuthoritiesByLogin(String login);
 
     <%_ if (authenticationType !== 'oauth2') { _%>
-    Mono<User> findOneWithAuthoritiesByEmailIgnoreCase(String email);
+    Mono<<%= asEntity('User') %>> findOneWithAuthoritiesByEmailIgnoreCase(String email);
 
     <%_ } _%>
     <%_ if (authenticationType == 'oauth2') { _%>
 
-    Mono<User> create(User user);
+    Mono<<%= asEntity('User') %>> create(<%= asEntity('User') %> user);
     <%_ } _%>
 
-    Flux<User> findAllWithAuthorities(Pageable pageable);
+    Flux<<%= asEntity('User') %>> findAllWithAuthorities(Pageable pageable);
 }
 
 class UserRepositoryInternalImpl implements UserRepositoryInternal {
@@ -310,19 +310,19 @@ class UserRepositoryInternalImpl implements UserRepositoryInternal {
     }
 
     @Override
-    public Mono<User> findOneWithAuthoritiesByLogin(String login) {
+    public Mono<<%= asEntity('User') %>> findOneWithAuthoritiesByLogin(String login) {
         return findOneWithAuthoritiesBy("login", login);
     }
 
     <%_ if (authenticationType !== 'oauth2') { _%>
     @Override
-    public Mono<User> findOneWithAuthoritiesByEmailIgnoreCase(String email) {
+    public Mono<<%= asEntity('User') %>> findOneWithAuthoritiesByEmailIgnoreCase(String email) {
         return findOneWithAuthoritiesBy("email", email.toLowerCase());
     }
 
     <%_ } _%>
     @Override
-    public Flux<User> findAllWithAuthorities(Pageable pageable) {
+    public Flux<<%= asEntity('User') %>> findAllWithAuthorities(Pageable pageable) {
         String property = pageable.getSort().stream().map(Sort.Order::getProperty).findFirst().get();
         String direction = String.valueOf(pageable.getSort().stream().map(Sort.Order::getDirection).findFirst().get());
         long page = pageable.getPageNumber();
@@ -332,7 +332,7 @@ class UserRepositoryInternalImpl implements UserRepositoryInternal {
             .sql("SELECT * FROM <%= jhiTablePrefix %>_user u LEFT JOIN <%= jhiTablePrefix %>_user_authority ua ON u.id=ua.user_id")
             .map(
                 (row, metadata) ->
-                    Tuples.of(r2dbcConverter.read(User.class, row, metadata), Optional.ofNullable(row.get("authority_name", String.class)))
+                    Tuples.of(r2dbcConverter.read(<%= asEntity('User') %>.class, row, metadata), Optional.ofNullable(row.get("authority_name", String.class)))
             )
             .all()
             .groupBy(t -> t.getT1().getLogin())
@@ -344,11 +344,11 @@ class UserRepositoryInternalImpl implements UserRepositoryInternal {
 
     <%_ if (authenticationType !== 'oauth2') { _%>
     @Override
-    public Mono<Void> delete(User user) {
+    public Mono<Void> delete(<%= asEntity('User') %> user) {
         return db.sql("DELETE FROM <%= jhiTablePrefix %>_user_authority WHERE user_id = :userId")
             .bind("userId", user.getId())
             .then()
-            .then(r2dbcEntityTemplate.delete(User.class)
+            .then(r2dbcEntityTemplate.delete(<%= asEntity('User') %>.class)
                 .matching(query(where("id").is(user.getId()))).all()
                 .then());
     }
@@ -356,17 +356,17 @@ class UserRepositoryInternalImpl implements UserRepositoryInternal {
     <%_ } _%>
     <%_ if (authenticationType === 'oauth2') { _%>
     @Override
-    public Mono<User> create(User user) {
-        return r2dbcEntityTemplate.insert(User.class).using(user).defaultIfEmpty(user);
+    public Mono<<%= asEntity('User') %>> create(<%= asEntity('User') %> user) {
+        return r2dbcEntityTemplate.insert(<%= asEntity('User') %>.class).using(user).defaultIfEmpty(user);
     }
 
     <%_ } _%>
-    private Mono<User> findOneWithAuthoritiesBy(String fieldName, Object fieldValue) {
+    private Mono<<%= asEntity('User') %>> findOneWithAuthoritiesBy(String fieldName, Object fieldValue) {
         return db.sql("SELECT * FROM <%= jhiTablePrefix %>_user u LEFT JOIN <%= jhiTablePrefix %>_user_authority ua ON u.id=ua.user_id WHERE u." + fieldName + " = :" + fieldName)
             .bind(fieldName, fieldValue)
             .map((row, metadata) ->
                 Tuples.of(
-                    r2dbcConverter.read(User.class, row, metadata),
+                    r2dbcConverter.read(<%= asEntity('User') %>.class, row, metadata),
                     Optional.ofNullable(row.get("authority_name", String.class))
                 )
             )
@@ -376,7 +376,7 @@ class UserRepositoryInternalImpl implements UserRepositoryInternal {
             .map(l -> updateUserWithAuthorities(l.get(0).getT1(), l));
     }
 
-    private User updateUserWithAuthorities(User user, List<Tuple2<User, Optional<String>>> tuples) {
+    private <%= asEntity('User') %> updateUserWithAuthorities(<%= asEntity('User') %> user, List<Tuple2<<%= asEntity('User') %>, Optional<String>>> tuples) {
         user.setAuthorities(
             tuples
                 .stream()
@@ -565,7 +565,7 @@ public class UserRepository {
         <%_ if (!reactive) { _%>
         return userDao.get(id);
         <%_ } else { _%>
-        return cqlTemplate.selectOneById(id, User.class)
+        return cqlTemplate.selectOneById(id, <%= asEntity('User') %>.class)
             .map(user -> {
                 if (user.getAuthorities() == null) {
                     user.setAuthorities(new HashSet<>());
@@ -605,7 +605,7 @@ public class UserRepository {
     <%_ if (!reactive) { _%>
         return userDao.findAll().all();
     <%_ } else { _%>
-        return cqlTemplate.select(findAllStmt.bind(), User.class)
+        return cqlTemplate.select(findAllStmt.bind(), <%= asEntity('User') %>.class)
             .map(user -> {
                 if (user.getAuthorities() == null) {
                     user.setAuthorities(new HashSet<>());

--- a/generators/server/templates/src/main/java/package/repository/rowmapper/UserRowMapper.java.ejs
+++ b/generators/server/templates/src/main/java/package/repository/rowmapper/UserRowMapper.java.ejs
@@ -30,7 +30,7 @@ import <%= packageName %>.service.ColumnConverter;
 import io.r2dbc.spi.Row;
 
 /**
- * Converter between {@link Row} to {@link User<%= entitySuffix %>}, with proper type conversions.
+ * Converter between {@link Row} to {@link <%= asEntity('User') %>}, with proper type conversions.
  */
 @Service
 public class UserRowMapper implements BiFunction<Row, String, <%= asEntity('User') %>> {

--- a/generators/server/templates/src/main/java/package/repository/rowmapper/UserRowMapper.java.ejs
+++ b/generators/server/templates/src/main/java/package/repository/rowmapper/UserRowMapper.java.ejs
@@ -24,16 +24,16 @@ import java.util.function.BiFunction;
 
 import org.springframework.stereotype.Service;
 
-import <%= packageName %>.domain.User;
+import <%= packageName %>.domain.<%= asEntity('User') %>;
 import <%= packageName %>.service.ColumnConverter;
 
 import io.r2dbc.spi.Row;
 
 /**
- * Converter between {@link Row} to {@link User}, with proper type conversions.
+ * Converter between {@link Row} to {@link User<%= entitySuffix %>}, with proper type conversions.
  */
 @Service
-public class UserRowMapper implements BiFunction<Row, String, User> {
+public class UserRowMapper implements BiFunction<Row, String, <%= asEntity('User') %>> {
     private final ColumnConverter converter;
     public UserRowMapper(ColumnConverter converter) {
         this.converter = converter;
@@ -41,11 +41,11 @@ public class UserRowMapper implements BiFunction<Row, String, User> {
 
     /**
      * Take a {@link Row} and a column prefix, and extract all the fields.
-     * @return the {@link User} stored in the database.
+     * @return the {@link <%= asEntity('User') %>} stored in the database.
      */
     @Override
-    public User apply(Row row, String prefix) {
-        User entity = new User();
+    public <%= asEntity('User') %> apply(Row row, String prefix) {
+        <%= asEntity('User') %> entity = new <%= asEntity('User') %>();
         entity.setId(row.get(prefix + "_id", <%= user.primaryKey.type %>.class));
         entity.setLogin(converter.fromRow(row, prefix + "_login", String.class));
         <%_ if (authenticationType !== 'oauth2') { _%>

--- a/generators/server/templates/src/main/java/package/service/UserService.java.ejs
+++ b/generators/server/templates/src/main/java/package/service/UserService.java.ejs
@@ -332,7 +332,7 @@ public class UserService {
             })
             .publishOn(Schedulers.boundedElastic())
             .then(Mono.fromCallable(() -> {
-                User newUser = new <%= asEntity('User') %>();
+                <%= asEntity('User') %> newUser = new <%= asEntity('User') %>();
                 <%_ if (databaseType === 'cassandra') { _%>
                 newUser.setId(UUID.randomUUID().toString());
                 <%_ } _%>
@@ -849,7 +849,7 @@ public class UserService {
     <%_ if (databaseType === 'sql') { _%>
     @Transactional
     <%_ } _%>
-    public Flux<User> removeNotActivatedUsersReactively() {
+    public Flux<<%= asEntity('User') %>> removeNotActivatedUsersReactively() {
         return userRepository
             .findAllByActivatedIsFalseAndActivationKeyIsNotNullAndCreatedDateBefore(<%_ if (databaseType === 'sql') { _%>LocalDateTime.ofInstant(Instant.now().minus(3, ChronoUnit.DAYS), ZoneOffset.UTC)<%_ } else { _%>Instant.now().minus(3, ChronoUnit.DAYS)<%_ } _%>)
             .flatMap(user -> userRepository.delete(user).thenReturn(user))

--- a/generators/server/templates/src/main/resources/config/liquibase/changelog/initial_schema.xml.ejs
+++ b/generators/server/templates/src/main/resources/config/liquibase/changelog/initial_schema.xml.ejs
@@ -134,6 +134,11 @@ _%>
                                  referencedTableName="<%= user.entityTableName %>"/>
     <%_ } _%>
     <%_ if (authenticationType !== 'oauth2' || (authenticationType === 'oauth2' && applicationType !== 'microservice')) { _%>
+        <% if (prodDatabaseType === 'mssql') { %>
+        <sql dbms="mssql">
+            SET IDENTITY_INSERT jhi_user ON;
+        </sql>
+        <% } %>    
         <<% if (prodDatabaseType === 'mssql') { %>ext:<% } %>loadData
                   file="config/liquibase/data/user.csv"
                   separator=";"
@@ -147,6 +152,11 @@ _%>
             <column name="activated" type="boolean"/>
             <column name="created_date" type="timestamp"/>
         </<% if (prodDatabaseType === 'mssql') { %>ext:<% } %>loadData>
+        <% if (prodDatabaseType === 'mssql') { %>
+        <sql dbms="mssql">
+            SET IDENTITY_INSERT jhi_user ON;
+        </sql>
+        <% } %>   
         <dropDefaultValue tableName="<%= user.entityTableName %>" columnName="created_date" columnDataType="${datetimeType}"/>
         <<% if (prodDatabaseType === 'mssql') { %>ext:<% } %>loadData
                   file="config/liquibase/data/authority.csv"

--- a/generators/server/templates/src/main/resources/config/liquibase/changelog/initial_schema.xml.ejs
+++ b/generators/server/templates/src/main/resources/config/liquibase/changelog/initial_schema.xml.ejs
@@ -139,44 +139,35 @@ _%>
             SET IDENTITY_INSERT jhi_user ON;
         </sql>
         <% } %>    
-        <<% if (prodDatabaseType === 'mssql') { %>ext:<% } %>loadData
+        <loadData
                   file="config/liquibase/data/user.csv"
                   separator=";"
-                  <%_ if (prodDatabaseType === 'mssql') { _%>
-                  identityInsertEnabled="true"
-                  <%_ } _%>
                   tableName="<%= user.entityTableName %>">
             <%_ if (prodDatabaseType === 'postgresql') { _%>
             <column name="id" type="<%= idField.loadColumnType %>"/>
             <%_ } _%>
             <column name="activated" type="boolean"/>
             <column name="created_date" type="timestamp"/>
-        </<% if (prodDatabaseType === 'mssql') { %>ext:<% } %>loadData>
+        </loadData>
         <% if (prodDatabaseType === 'mssql') { %>
         <sql dbms="mssql">
             SET IDENTITY_INSERT jhi_user OFF;
         </sql>
         <% } %>   
         <dropDefaultValue tableName="<%= user.entityTableName %>" columnName="created_date" columnDataType="${datetimeType}"/>
-        <<% if (prodDatabaseType === 'mssql') { %>ext:<% } %>loadData
+        <loadData
                   file="config/liquibase/data/authority.csv"
                   separator=";"
-                  <%_ if (prodDatabaseType === 'mssql') { _%>
-                  identityInsertEnabled="true"
-                  <%_ } _%>
                   tableName="<%= jhiTablePrefix %>_authority">
             <column name="name" type="string"/>
-        </<% if (prodDatabaseType === 'mssql') { %>ext:<% } %>loadData>
+        </loadData>
 
-        <<% if (prodDatabaseType === 'mssql') { %>ext:<% } %>loadData
+        <loadData
                   file="config/liquibase/data/user_authority.csv"
                   separator=";"
-                  <%_ if (prodDatabaseType === 'mssql') { _%>
-                  identityInsertEnabled="true"
-                  <%_ } _%>
                   tableName="<%= jhiTablePrefix %>_user_authority">
             <column name="user_id" type="<%= idField.loadColumnType %>"/>
-        </<% if (prodDatabaseType === 'mssql') { %>ext:<% } %>loadData>
+        </loadData>
     <%_ } _%>
 <%_ } _%>
     </changeSet>

--- a/generators/server/templates/src/main/resources/config/liquibase/changelog/initial_schema.xml.ejs
+++ b/generators/server/templates/src/main/resources/config/liquibase/changelog/initial_schema.xml.ejs
@@ -154,7 +154,7 @@ _%>
         </<% if (prodDatabaseType === 'mssql') { %>ext:<% } %>loadData>
         <% if (prodDatabaseType === 'mssql') { %>
         <sql dbms="mssql">
-            SET IDENTITY_INSERT jhi_user ON;
+            SET IDENTITY_INSERT jhi_user OFF;
         </sql>
         <% } %>   
         <dropDefaultValue tableName="<%= user.entityTableName %>" columnName="created_date" columnDataType="${datetimeType}"/>

--- a/generators/server/templates/src/test/java/package/web/rest/UserResourceIT.java.ejs
+++ b/generators/server/templates/src/test/java/package/web/rest/UserResourceIT.java.ejs
@@ -257,7 +257,7 @@ class UserResourceIT <% if (databaseType === 'cassandra') { %>extends AbstractCa
     public static void deleteEntities(EntityManager em) {
         try {
             em.deleteAll("<%= jhiTablePrefix %>_user_authority").block();
-            em.deleteAll(User.class).block();
+            em.deleteAll(<%= asEntity('User') %>.class).block();
         } catch (Exception e) {
             // It can fail, if other entities are still referring this - it will be removed later.
         }

--- a/test/app.spec.js
+++ b/test/app.spec.js
@@ -416,7 +416,14 @@ describe('JHipster generator', () => {
         assert.file(expectedFiles.mssql);
         assert.file(expectedFiles.hibernateTimeZoneConfig);
         assert.fileContent('pom.xml', /mssql-jdbc/);
-        assert.fileContent(`${SERVER_MAIN_RES_DIR}config/liquibase/changelog/00000000000000_initial_schema.xml`, /identityInsertEnabled/);
+        assert.fileContent(
+          `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/00000000000000_initial_schema.xml`,
+          /SET IDENTITY_INSERT jhi_user ON;/
+        );
+        assert.fileContent(
+          `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/00000000000000_initial_schema.xml`,
+          /SET IDENTITY_INSERT jhi_user OFF;/
+        );
       });
       shouldBeV3DockerfileCompatible('mssql');
     });

--- a/test/generator-base.spec.js
+++ b/test/generator-base.spec.js
@@ -738,9 +738,9 @@ describe('Generator Base', () => {
       });
     });
     describe('when called for mssql', () => {
-      it('return r2dbc:sqlserver://localhost:1433;database=test', () => {
+      it('return r2dbc:mssql://localhost:1433/test', () => {
         expect(BaseGenerator.getR2DBCUrl('mssql', { databaseName: 'test', hostname: 'localhost' })).to.equal(
-          'r2dbc:sqlserver://localhost:1433;database=test'
+          'r2dbc:mssql://localhost:1433/test'
         );
       });
     });

